src/analytics/lambdas/scan-metadata-workflow/scan-metadata.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import { logger } from '../../../common/powertools';
+import { ScanMetadataBody } from '../../private/model';
+import { getRedshiftClient, executeStatements, getRedshiftProps } from '../redshift-data';
+
+const REDSHIFT_DATA_API_ROLE_ARN = process.env.REDSHIFT_DATA_API_ROLE!;
+const REDSHIFT_DATABASE = process.env.REDSHIFT_DATABASE!;
+
+// Create an Amazon service client object.
+const redshiftDataApiClient = getRedshiftClient(REDSHIFT_DATA_API_ROLE_ARN);
+
+export interface ScanMetadataEvent {
+  detail: ScanMetadataBody;
+}
+
+/**
+ * The lambda function submit a SQL statement to scan metadata.
+ * @param event ScanMetadataEvent, the JSON format is as follows:
+ {
+    "detail": {
+      "appId": app1
+    }
+  }
+  @returns The query_id and relevant properties.
+ */
+export const handler = async (event: ScanMetadataEvent) => {
+  const redshiftProps = getRedshiftProps(
+    process.env.REDSHIFT_MODE!,
+    REDSHIFT_DATABASE,
+    REDSHIFT_DATA_API_ROLE_ARN,
+    process.env.REDSHIFT_DB_USER!,
```

Comment:
`DB_USER` and `CLUSTER_IDENTIFIER` are for provisioned redshift, `WORKGROUP` is for Redshift serverless, I'm wondering why all of them have values?

Reply:
Yes, they do not have the value in the same case.
We set the values in scan-metadata-workflow.ts according to redshift mode.
The process the same with upsert-user-workflow.

Reply:
Yes, it is safe.
We have below logic to set these env var. Make sure DB_USER and CLUSTER_IDENTIFIER value is '' for SERVERLESS mode and WORKGROUP value is '' for PROVISIONED mode.
https://github.com/awslabs/clickstream-analytics-on-aws/blob/scan-metadata/src/analytics/private/scan-metadata-workflow.ts#L199

If not `!`, there will be below error:
`Argument of type 'string | undefined' is not assignable to parameter of type 'string'.`
Because existing function getRedshiftProps define the input parameter as below:
```
(
  redshiftMode: string,
  databaseName: string,
  dataAPIRoleArn: string,
  dbUser: string,
  workgroupName: string,
  clusterIdentifier: string,
)
```